medor_website_product_subscription/controllers/main.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+from openerp import http
+from openerp.http import request
+from openerp.tools.translate import _
+
+from openerp.exceptions import ValidationError
+
+from openerp.addons.medor_website_product_subscription.controllers.medor_form import UserForm
+
+
+class MedorWebsiteProductSubscription(http.Controller):
+
+    @http.route(
+        '/edit/user/',
+        type='http',
+        auth='user',
+        website=True
+    )
+    def user_form(self, **kw):
+        user = request.env.user
+        form = UserForm(request.params, user=request.env.user)
+        form.normalize_form_data()
+        form.validate_form()
+        form.init_form_data()
+        form.set_form_defaults()
+        if ('error' not in request.params
+                and request.httprequest.method == 'POST'):
+            # representative
+            values = {
+                key: request.params[key]
+                for key in request.params
+                if key in form.user_fields
+            }
+            user.write(values)
+            if user.parent_id:
```

Comment:
from my knowledge user doesn't have a parent_id

Reply:
@remytms Yep, I think I said in the PR I closed that it should be writing on `user.partner_id`.

Reply:
👍

Reply:
`user` has a `parent_id` field. If I remember well `user` inherit from `partner`.

Reply:
^ ce génie

```python
class res_users(osv.osv):
    """ User class. A res.users record models an OpenERP user and is different
        from an employee.

        res.users class now inherits from res.partner. The partner model is
        used to store the data related to the partner: lang, name, address,
        avatar, ... The user model is now dedicated to technical data.
    """
    __uid_cache = {}
    _inherits = {
        'res.partner': 'partner_id',
    }
```